### DISCUSSION TITLE
Help needed: Synology / ebusd / adapter v5 -> ebusd connected: no

### DISCUSSION THREAD

**Discussion by Humpelman** [john30/ebusd#1004] [Q&A]
Dear all,

after hours of trying I'm still having trouble getting the ebus adapter v5 to work on my synology DS220+ (Docker/ioBorker + Docker/ebusd).

The ebus adapter is connected directly to the Synology via USB and is unfortunately not recognized by it, at least it is not listed in the Synology among the devices.

With easi I switched the ebus to "enhanved via USB" so that I get "ebusd device string: ens:/dev/ttyACM0 (number may vary)" in the title bar.

On the DS220+, "john30/ebusd:latest" runs in a Docker container and I tried "EBUSD_OPTS="--scanconfig"" and "EBUSD_OPTS="--scanconfig -d ens:/dev/ttyACM0"" as parameters.

All without success, no connection to ebusd:

Chip ID: dc547563d668, ESP32-C3, rev 4
Hostname: ebus-63d668
PCB: 20.0
Flash: 4194304
Version: v5.2-dev-2164-g3befd5fff7
Reset reason: 5
eBUS r/w: 59536/0
Host r/w: 0/0
Up time: 1471
Free heap: 124980
ebusd connected: no
eBUS signal: acquired

Could you help me, please?
At the end of the setup I would like to be able to write and read on my Vaillant heating system.

**Comment by Koky05** [comment]
For Synology DSM 7.x you need driver for any devices you connect to it. In these case you need to create driver by your self it does not exists. Then when it recognize it it could by connected to docker container as '--devices /dev/ttyACM0' parameter.
All this was reason I use WiFi version with mine DS720+.

You can try to contact Robert Klep who creates several drivers for Synology NAS:
[dsm7-usb-serial-drivers](https://github.com/robertklep/dsm7-usb-serial-drivers/commits?author=robertklep)

**Comment by Humpelman** [comment]
Thanks for your feedback!
Just after my post I got with "ebus: enhanved via RXTX" a "ebusd connected: yes (inactive)", but this was a deadend aswell. I was not able to get to the ebusd html page ...

How did you realized the WiFi connection?

easi/Configuration/ebus: enhanced via TCP

Docker/ebusd-Container started via putty: sudo docker run -d --name=ebusd -p 8889:8889 -e EBUSD_OPTS="--scanconfig -d ens:192.168.178.112:9999" john30/ebusd

With this setting I get "ebusd connected: no" again.

**Comment by Koky05** [comment]
I use these docker-compose config alse to read data I use MQTT with Home Assistant autodiscovery :

> services:
>   ebusd:
>     image: john30/ebusd:latest
>     container_name: ebusd
>     restart: unless-stopped
>     network_mode: host
>     environment:
>       - TZ=Europe/Bratislava
>       - EBUSD_DEVICE=ens:192.168.178.112:9999   #Changed to your IP address of eBus Adapter
>       - EBUSD_ARCH=amd64
>       - EBUSD_SCANCONFIG=full
>       - EBUSD_CONFIGPATH=/etc/ebusd/ebusd-configuration/ebusd-2.1.x/en  #I use local copy of configuration
>       - EBUSD_CONFIGLANG=EN
>       - EBUSD_LOGAREAS=all
>       - EBUSD_LOGLEVEL=error
>       - EBUSD_ACCESSLEVEL=*
>       - EBUSD_MQTTHOST=127.0.0.1  #use IP address of MQTT broker
>       - EBUSD_MQTTPORT=1883         #Port of MQTT broker
>       - EBUSD_MQTTUSER=${MQTT_USER}
>       - EBUSD_MQTTPASS=${MQTT_PASSWORD}
>       - EBUSD_MQTTTOPIC=ebusd
>       - EBUSD_MQTTJSON=
>       - EBUSD_MQTTINT=/etc/ebusd/mqtt-hassio.cfg
>       - EBUSD_MQTTVERBOSE=
>       - EBUSD_MQTTRETAIN=
>       - EBUSD_MQTTCHANGES=
>     volumes:
>       - "/etc/localtime:/etc/localtime:ro"
>       - "/volume1/docker/ebusd/data:/data"
>       - "/volume1/docker/ebusd:/ebusd"
>       - "/volume1/docker/ebusd/config:/etc/ebusd"
>       - "/volume1/docker/ebusd/log:/var/log"
>

**Reply by Humpelman** [reply]
Finally ebusd is connected to iobroker. I used sudo docker run -d --name=ebusd -p 8888:8888 john30/ebusd --scanconfig -d ens:192.168.178.112:9999 --mqttport=1883 --mqtthost=192.168.178.104 for this.
But now I only get the outside temperature and I´m struggeling to get all the other values.

I once again have suspicions about Synology as to why the whole thing isn't going as well as hoped.
How did you install your MQTT client, where should I add the code mentioned in the wiki? https://github.com/john30/ebusd/wiki/3.3.-MQTT-client

**Reply by Koky05** [reply]
For MQTT client I use next config in docker-compose file:

>   #MQTT
>services:
>   mqtt:
>     container_name: mqtt 
>     image: eclipse-mosquitto:openssl
>     command: /usr/sbin/mosquitto -c /mosquitto/config/mosquitto.conf
>     network_mode: host
>     environment:
>       TZ: 'Europe/Bratislava'
>     volumes:
>       - "/volume1/docker/mqtt/log:/mosquitto/log"
>       - "/volume1/docker/mqtt/config/mosquitto.conf:/mosquitto/config/mosquitto.conf"
>       - "/volume1/docker/mqtt/config:/mosquitto/config"
>       - "/volume1/docker/mqtt/data:/mosquitto/data"
>     labels: 
>       - "com.centurylinklabs.watchtower.enable=true"
>     restart: unless-stopped

But you can find lost of manuals on internet mine favorite if BeardedThinker on youtube:
[BeardedTinker](https://www.youtube.com/@BeardedTinker/search?query=Synology)
